Sources/SpeziAccessGuard/AccessGuard.swift

```diff
@@ -81,7 +81,7 @@ public final class AccessGuard {
         self.configurations = configurations
     }
     
-    
+    @MainActor
```

Comment:
This already schedules the work to run on the main thread. There is no need to annotate this. The same thing goes for the other functions too.

Reply:
Thank you for the feedback, initially I annotated the function with `@MainActor` as without it would cause the error "_Task or actor isolated value cannot be sent_". After further investigation of the underlying issue I updated AccessGuard to be @Sendable, in this manner the function can be called from any context as before. If you agree with this change, I will add the necessary documentation!

Reply:
I am not sure why these changes are necessary 🤔
The code that you are changing already runs on the Main Thread.

Reply:
Or we can have something like this: 

```swift 
@MainActor
func sceneDidEnterBackground(_ scene: UIScene) {
    inTheBackground = true
    lastEnteredBackground = .now

    for viewModel in viewModels.values {
        viewModel.didEnterBackground()
    }
}

@MainActor
    func sceneWillEnterForeground(_ scene: UIScene) {
    inTheBackground = false

    for viewModel in viewModels.values {
        viewModel.willEnterForeground(lastEnteredBackground: lastEnteredBackground)
    }
}
```

Reply:
I think `LifecycleHandler` in general is deprecated at this point; we might want to consider that as well. I think making this Main Actor isolated is not a bad thing.